Panic empty `clouds.yaml` file

```
--- FAIL: TestObsBucketLifecycle (0.00s)
panic: assignment to entry in nil map [recovered]
        panic: assignment to entry in nil map

goroutine 19 [running]:
testing.tRunner.func1.2(0x765920, 0x830c70)
        /snap/go/current/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc000082900)
        /snap/go/current/src/testing/testing.go:1146 +0x4b6
panic(0x765920, 0x830c70)
        /snap/go/current/src/runtime/panic.go:965 +0x1b9
github.com/opentelekomcloud/gophertelekomcloud/openstack.(*Env).loadOpenstackConfig(0xc0000d6540, 0x7fe05c762fff, 0xc00004ca48, 0x41861b)
        /home/akachurin/GolandProjects/gophertelekomcloud/openstack/loader.go:556 +0x585
github.com/opentelekomcloud/gophertelekomcloud/openstack.(*Env).Cloud(0xc0000d6540, 0x0, 0x0, 0x0, 0x27, 0xc00003a508, 0x459850)
        /home/akachurin/GolandProjects/gophertelekomcloud/openstack/loader.go:478 +0x88
github.com/opentelekomcloud/gophertelekomcloud/acceptance/clients.CloudAndClient(0x438, 0x4f70f3, 0x43800000007)
        /home/akachurin/GolandProjects/gophertelekomcloud/acceptance/clients/clients.go:541 +0x45
github.com/opentelekomcloud/gophertelekomcloud/acceptance/clients.NewOBSClient(0x4f70c0, 0x982be0, 0x9ceb60)
        /home/akachurin/GolandProjects/gophertelekomcloud/acceptance/clients/clients.go:276 +0x37
github.com/opentelekomcloud/gophertelekomcloud/acceptance/openstack/obs/v1.TestObsBucketLifecycle(0xc000082900)
        /home/akachurin/GolandProjects/gophertelekomcloud/acceptance/openstack/obs/v1/obs_test.go:15 +0x34
testing.tRunner(0xc000082900, 0x7e40e8)
        /snap/go/current/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
        /snap/go/current/src/testing/testing.go:1238 +0x2b3
```

--- a/openstack/loader.go
+++ b/openstack/loader.go
@@ -184,6 +184,12 @@ type Config struct {
 	Clouds       map[string]Cloud `yaml:"clouds" json:"clouds"`
 }
 
+func NewConfig() *Config {
+	return &Config{
+		Clouds: map[string]Cloud{},
+	}
+}
+
 // AuthType represents a valid method of authentication: `password`, `token`, `aksk` or `agency`
 type AuthType string
 
@@ -337,7 +343,7 @@ func loadCloudFile(path string) (*Config, error) {
 	if err != nil {
 		return nil, err
 	}
-	clouds := new(Config)
+	clouds := NewConfig()
 	if err := yaml.Unmarshal(data, clouds); err != nil {
 		return nil, err
 	}
@@ -525,9 +531,7 @@ func (e *Env) loadOpenstackConfig() (*Config, error) {
 	}
 	vendorPath := selectExisting(vendors)
 
-	cloudConfig := &Config{
-		Clouds: map[string]Cloud{},
-	}
+	cloudConfig := NewConfig()
 
 	// load clouds.yaml
 	if configPath != "" {

--- a/openstack/loader_test.go
+++ b/openstack/loader_test.go
@@ -134,6 +134,22 @@ func TestCloudYamlPaths(t *testing.T) {
 	}
 }
 
+func TestEmptyClouds(t *testing.T) {
+	_ = os.Setenv("OS_CLOUD", "useless_cloud")
+	_ = os.Setenv("OS_CLIENT_CONFIG_FILE", "./clouds.yaml")
+	cwd, _ := os.Getwd()
+
+	fileName := "clouds.yaml"
+	currentConfigDir := filepath.Join(cwd, fileName)
+	files := []string{currentConfigDir}
+	backupFiles(t, currentConfigDir)
+	defer restoreBackup(t, files...)
+
+	th.AssertNoErr(t, ioutil.WriteFile(fileName, []byte{}, 0644))
+	_, err := NewEnv("OS_").Cloud()
+	th.AssertNoErr(t, err)
+}
+
 func TestCloudName(t *testing.T) {
 	_ = os.Setenv("OS_CLOUD", tools.RandomString("CLD_", 5))
 	expectedName := tools.RandomString("CLD_SET_", 5)

